drivers/at86rf2xx/at86rf2xx_getset.c

```diff
@@ -125,9 +125,9 @@ uint64_t at86rf2xx_get_addr_long(at86rf2xx_t *dev)
 void at86rf2xx_set_addr_long(at86rf2xx_t *dev, uint64_t addr)
 {
     for (int i = 0; i < 8; i++) {
-        dev->addr_long[i] = (addr >> ((7 - i) * 8));
+        dev->long_addr[i] = (uint8_t)(addr >> (7 - i));
         at86rf2xx_reg_write(dev, (AT86RF2XX_REG__IEEE_ADDR_0 + i),
-                            dev->addr_long[i]);
+                            (addr >> i));
```

Comment:
Not quite sure, but with this I get the same long address on every node. Reverting this patch, gives me at least different long addresses.